contracts/src/internal/HyperdriveBase.sol

```diff
@@ -682,6 +677,152 @@ abstract contract HyperdriveBase is IHyperdriveEvents, HyperdriveStorage {
+        int256 _shareAdjustment,
+        uint256 _bondReserves,
+        uint256 _vaultSharePrice,
+        uint256 _longExposure,
+        int256 _checkpointExposure
+    ) internal view returns (int256, bool) {
+        // Calculate the share payment and bond proceeds of opening the largest
+        // possible long on the YieldSpace curve. This does not include fees.
+        // These calculations fail when the max long is close to zero, and we
+        // ignore these failures since we can proceed with the calculation in
+        // this case.
+        (uint256 effectiveShareReserves, bool success) = HyperdriveMath
+            .calculateEffectiveShareReservesSafe(
+                _shareReserves,
+                _shareAdjustment
+            );
+        if (!success) {
+            return (0, false);
+        }
+        (uint256 maxSharePayment, ) = YieldSpaceMath
+            .calculateMaxBuySharesInSafe(
+                effectiveShareReserves,
+                _bondReserves,
+                ONE - _timeStretch,
+                _vaultSharePrice,
+                _initialVaultSharePrice
+            );
+        (uint256 maxBondProceeds, ) = YieldSpaceMath
+            .calculateBondsOutGivenSharesInDownSafe(
+                effectiveShareReserves,
+                _bondReserves,
+                maxSharePayment,
+                ONE - _timeStretch,
+                _vaultSharePrice,
+                _initialVaultSharePrice
+            );
+
+        // If one of the max share payment or max bond proceeds calculations
+        // fail or return zero, the max long amount is zero plus or minus a few
+        // wei.
+        if (maxSharePayment == 0 || maxBondProceeds == 0) {
+            maxSharePayment = 0;
+            maxBondProceeds = 0;
+        }
+
+        // Apply the fees from opening a long to the max share payment and bond
+        // proceeds. Fees can impact solvency, so it's important to account for
+        // them here.
+        uint256 spotPrice = HyperdriveMath.calculateSpotPrice(
+            effectiveShareReserves,
+            _bondReserves,
+            _initialVaultSharePrice,
+            _timeStretch
+        );
+        (maxSharePayment, maxBondProceeds, ) = _calculateOpenLongFees(
+            maxSharePayment,
+            maxBondProceeds,
+            _vaultSharePrice,
+            spotPrice
+        );
+
+        // Calculate the pool's solvency after opening the max long.
+        uint256 shareReserves = _shareReserves + maxSharePayment;
+        uint256 longExposure = LPMath.calculateLongExposure(
+            _longExposure,
+            _checkpointExposure,
+            _checkpointExposure + maxBondProceeds.toInt256()
+        );
+        uint256 vaultSharePrice = _vaultSharePrice;
+        return (
+            shareReserves.mulDown(vaultSharePrice).toInt256() -
+                longExposure.toInt256() -
+                _minimumShareReserves.mulUp(vaultSharePrice).toInt256(),
```

Comment:
this was 2*z_min before, what did you do to make this work without the padding?

Reply:
Let’s talk about that tomorrow. I suspect it has to do with the improvements we made to the invariant check, but I was actually somewhat confused about why we needed it before since all of the tests passed without it (unless I’m missing something)

Reply:
it's possible the inclusion of the fees into the calc fixed it?